src/hammer-vlsi/hammer_driver.py

```diff
@@ -148,6 +153,80 @@ def update_tool_configs(self) -> None:
         tools = reduce(lambda a, b: a + b, list(self.tool_configs.values()))
         self.database.update_tools(tools)
 
+    def load_formal_tool(self, run_dir: str = "") -> bool:
+        """
+        Load the place and route tool based on the given database.
+
+        :param run_dir: Directory to use for the tool run_dir. Defaults to the run_dir passed in the HammerDriver
+                        constructor.
+        """
+        if run_dir == "":
+            # Use the synthesis directory for formal tools
+            run_dir = os.path.join(self.obj_dir, "syn-rundir")
+
+        formal_tool_name = self.database.get_setting("vlsi.core.formal_tool")
+        formal_tool_get = load_tool(
+            path=self.database.get_setting("vlsi.core.formal_tool_path"),
+            tool_name=formal_tool_name
+        )
+        assert isinstance(formal_tool_get, HammerFormalTool), "Formal tool must be a HammerFormalTool"
+        formal_tool = formal_tool_get  # type: HammerTool
+        formal_tool.name = formal_tool_name
+        formal_tool.logger = self.log.context("formal")
+        formal_tool.set_database(self.database)
+        formal_tool.run_dir = run_dir
+
+        formal_tool.input_files = self.database.get_setting("synthesis.inputs.input_files")
+        if len(formal_tool.input_files) < 2:
+            self.log.error("There must be at least two input files for formal tools: reference and implementation")
+            return False
+
+        self.formal_tool = formal_tool
+
+        self.tool_configs["formal"] = formal_tool.get_config()
+        self.update_tool_configs()
+        return True
+
+    def load_power_tool(self, run_dir: str = "") -> bool:
+        """
+        Load the place and route tool based on the given database.
+
+        :param run_dir: Directory to use for the tool run_dir. Defaults to the run_dir passed in the HammerDriver
+                        constructor.
+        """
+        if run_dir == "":
+            run_dir = os.path.join(self.obj_dir, "pwr-rundir")
+
+        power_tool_name = self.database.get_setting("vlsi.core.power_tool")
+        power_tool_get = load_tool(
+            path=self.database.get_setting("vlsi.core.power_tool_path"),
+            tool_name=power_tool_name
+        )
+        assert isinstance(power_tool_get, HammerPowerTool), "Power tool must be a HammerPowerTool"
+        power_tool = power_tool_get  # type: HammerPowerTool
+        power_tool.name = power_tool_name
+        power_tool.logger = self.log.context("power")
+        power_tool.technology = self.tech
+        power_tool.set_database(self.database)
+        power_tool.run_dir = run_dir
+
+        # TODO: better way to get input files?
```

Comment:
I'll leave @edwardcwang to comment on a better way to do this. I assume its a similar parallel to `par.inputs`?